app/controllers/mixins/generic_show_mixin.rb

```diff
@@ -107,6 +107,26 @@ def display_nested_generic(display)
       nested_list(display, display.camelize.singularize.constantize)
     end
 
+    def display_descendant_vms
+      @showtype = "config"
+      drop_breadcrumb(:name => _("%{name} (All VMs - Tree View)") % {:name => @record.name},
+                      :url  => "/#{controller_name}/show/#{@record.id}?display=descendant_vms&treestate=true")
```

Comment:
I moved a helper method `show_link` that should be useful for generating the URL here. Also if you use that method, the code will work w/o changes when controllers are changed from `restful? == false` to `restful? == true`.